lib/dl_formula/dl_formula/mutation/lod.py

```diff
@@ -29,21 +26,19 @@ class ExtAggregationToQueryForkMutation(DimensionResolvingMutationBase):
     """
 
     def match_node(self, node: nodes.FormulaItem, parent_stack: Tuple[nodes.FormulaItem, ...]) -> bool:
-        is_agg = is_aggregate_function(node)
-        direct_parent = parent_stack[-1]
-        already_patched = (
-            isinstance(direct_parent, fork_nodes.QueryFork)
-            and direct_parent.result_expr is node
-            and qfork_is_aggregation(direct_parent)  # is an aggregation fork (vs. AGO forks)
-        )
-        return is_agg and not already_patched
+        return is_aggregate_function(node)
 
     def make_replacement(
         self, old: nodes.FormulaItem, parent_stack: Tuple[nodes.FormulaItem, ...]
     ) -> nodes.FormulaItem:
         assert isinstance(old, nodes.FuncCall)
 
-        dimensions, _, parent_dimension_set = self._generate_dimensions(node=old, parent_stack=parent_stack)
+        dimensions: list[nodes.FormulaItem]
+        if old.lod.list_node_type(aux_nodes.ErrorNode):
+            # there are errors in current LODs, propagate them
+            dimensions = list(old.lod.children)
```

Comment:
After changing the order of modifications, it is possible that we can suppress errors from children's LODs. To avoid this, we explicitly check for inner errors and propagate them if they exist